Frontend/auction-app/src/pages/Authentication/Registration.jsx

```diff
@@ -0,0 +1,42 @@
+import React from "react";
+import GenericField from "../../Components/FormField/GenericField"
+import { handleFieldChange } from "../index.jsx"
+import routes from "../../Util/routes"
+import axios from "axios"
+
+
+const Registration = props => {
+    const [user, setUser] = React.useState({
+        firstName: "",
+        lastName: "",
+        email: "",
+        password: ""
+    })
+
+    return (
+        <div>
+            <GenericField id={"firstName"} name={"firstName"} label={"First Name"} type={"text"} onChange={(e) => handleFieldChange(e, user, setUser)} />
+            <GenericField id={"lastName"} name={"lastName"} label={"Last Name"} type={"text"} onChange={(e) => handleFieldChange(e, user, setUser)} />
+            <GenericField id={"email"} name={"email"} label={"Email"} type={"text"} onChange={(e) => handleFieldChange(e, user, setUser)} />
+            <GenericField id={"password"} name={"password"} label={"Password"} type={"text"} onChange={(e) => handleFieldChange(e, user, setUser)} />
+            <button type={"button"} onClick={() => handleRegisterClick(user)} >REGISTER</button>
+        </div>
+    )
+
+    function handleRegisterClick(user) {
+        //TODO: Find other way for sending role id
+        user.roleId = 2
```

Comment:
I would suggest not sending roleId at all. The registration api should only register users. (role = 2)
So the roleid can be set in the backend while creating the model object. (the DTO does not need to the contain roleid field)